src-self-hosted/test.zig

```diff
@@ -415,26 +413,25 @@ pub const TestContext = struct {
             progress.initial_delay_ns = 0;
             progress.refresh_rate_ns = 0;
 
-            const info = try std.zig.system.NativeTargetInfo.detect(std.testing.allocator, case.target);
-            try self.runOneCase(std.testing.allocator, &prg_node, case, info.target);
+            try self.runOneCase(std.testing.allocator, &prg_node, case);
             try std.testing.allocator_instance.validate();
         }
     }
 
-    fn runOneCase(self: *TestContext, allocator: *Allocator, root_node: *std.Progress.Node, case: Case, target: std.Target) !void {
+    fn runOneCase(self: *TestContext, allocator: *Allocator, root_node: *std.Progress.Node, case: Case) !void {
         var tmp = std.testing.tmpDir(.{});
         defer tmp.cleanup();
 
         const tmp_src_path = if (case.extension == .Zig) "test_case.zig" else if (case.extension == .ZIR) "test_case.zir" else unreachable;
         const root_pkg = try Package.create(allocator, tmp.dir, ".", tmp_src_path);
         defer root_pkg.destroy();
 
-        const bin_name = try std.zig.binNameAlloc(allocator, "test_case", target, case.output_mode, null);
+        const bin_name = try std.zig.binNameAlloc(allocator, "test_case", case.target.toTarget(), case.output_mode, null);
```

Comment:
```zig
    /// TODO deprecated, use `std.zig.system.NativeTargetInfo.detect`.
    pub fn toTarget(self: CrossTarget) Target {
        return .{
            .cpu = self.getCpu(),
            .os = self.getOs(),
            .abi = self.getAbi(),
        };
    }
```

Reply:
Yeah, looking more closely at the diff for that commit made it obvious what I was doing wrong.

Good now.